vendor/github.com/ethereum/go-ethereum/whisper/whisperv5/peer.go

```diff
@@ -32,7 +33,7 @@ type Peer struct {
 	host    *Whisper
 	peer    *p2p.Peer
 	ws      p2p.MsgReadWriter
-	trusted bool
+	trusted int32
```

Comment:
I don't think it's a good idea to make something `int32` just because we want to use `atomic`. If all other methods of `Peer` are not thread-safe, we should not make an exception here. If atomicity is required, I would go with a mutex.

Reply:
only 2 fields can be changed on Peer structure:
- known, and it is a sync.Map
- trusted wasn't protected
imo atomic is much cleaner here

Reply:
Maybe a cleaner alternative could be `atomic.Value` and checking it for `nil` or `!nil`.

Reply:
folks it is common use case across go-ethereum, whenever there is a need for thread-safe bool atomic integer is used

Reply:
Well, it is always possible to wrap `int32` into something like `type AtomicBool int32` and has `IsTrue()` and `IsFalse()` values.
Or make an enum.
In general, I heavily dislike using `bool`s for state-machines, it is not future proof. Imagine you have to add one more state, and that's impossible by a bool, you can only make 2^(n+1) states.
Always having enums might seems like an overkill, but it helps getting control over states and avoid lots of bugs caused by a weird combination of bools.

Reply:
folks, this is a simple change that is consistent with go-ethereum codebase, and common golang practice in general. i like the atomic bool but i don't want to add this object to use it in single place